license-check.sh

```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env bash
+
+files=`find . -type f -name '*.java' | xargs grep -L "Licensed under the Apache License"`
+
+if [[ -z ${files} ]]
+then
+	echo "No files missing headers."
+	exit 0
+else
+	echo "Consider running header-update.sh from the root to update these files automatically!"
```

Comment:
Would it be better to swap lines 10 and 11 around?

Reply:
Line 11 precedes the list of failed commits (loop 14-16). I think you are correct in essence though - moving line 10 to after the loop seems very sensible.